### ISSUE TITLE
Node.js 15 failures

### ISSUE THREAD

**Issue by BethGriggs** [nodejs/citgm#836]
- **Node Version**: v15.0.0
- **CitGM Version**: v7.1.9

**All platforms**:
- [x] citgm.@hapi/shot-v5.0.4 - https://github.com/hapijs/lab/pull/997
- [x] citgm.resolve-v1.17.0 - needs update for new core modules
- [ ] citgm.serialport-v9.0.2 - npm 7,  should be fixed by https://github.com/serialport/node-serialport/pull/2153
- [ ] citgm.tape-v5.0.1 - fixed by 1ac9ecf927bb1958857cb4050ec725cd3f22d27a to handle stack changes.
- [ ] citgm.ws-v7.3.1 - ws issue is fixed in npm@7.0.4
- [ ] citgm.yargs-v16.1.0 - npm 7, fix in progress

**macOS**:
- osx10.14 and osx10.15 have a bunch of additional failures, but they do seem familiar.
  - https://ci.nodejs.org/view/Node.js-citgm/job/citgm-smoker/2523/nodes=osx1014/testReport

**Comment by ljharb** [comment]
What are the failures with tape? Which core modules is resolve failing to detect?

**Comment by BethGriggs** [comment]
tape@5.0.1 from https://ci.nodejs.org/view/Node.js-citgm/job/citgm-smoker/2523/nodes=ubuntu1804-64/testReport/(root)/citgm/tape_v5_0_1/:

<details>

```
        not ok 3 - should be equivalent
           ---
           found: |
             $TEST/async-await/sync-error.js:7
                 throw new Error('oopsie');
                 ^
             Error: oopsie
                 at Test.myTest ($TEST/async-await/sync-error.js:$LINE:$COL)
                 at Test.bound [as _cb] ($TAPE/lib/test.js:$LINE:$COL)
                 at Test.run ($TAPE/lib/test.js:$LINE:$COL)
                 at Test.bound [as run] ($TAPE/lib/test.js:$LINE:$COL)
                 at processImmediate (node:internal/timers:$LINE:$COL)
           wanted: |
             $TEST/async-await/sync-error.js:7
                 throw new Error('oopsie');
                 ^
             Error: oopsie
                 at Test.myTest ($TEST/async-await/sync-error.js:$LINE:$COL)
                 at Test.bound [as _cb] ($TAPE/lib/test.js:$LINE:$COL)
                 at Test.run ($TAPE/lib/test.js:$LINE:$COL)
                 at Test.bound [as run] ($TAPE/lib/test.js:$LINE:$COL)
           at:
             line: 189
             column: 11
             file: test/async-await.js
           stack: |
             test/async-await.js:189:11
             ChildProcess.<anonymous> (test/common.js:90:9)
             ChildProcess.emit (node:events:327:20)
             Process.ChildProcess._handle.onexit (node:internal/child_process:277:12)
           source: |
             t.same(stripFullStack(stderr), [
           ...
         1..3
         # failed 1 of 3 tests
     not ok 6 - sync-error # time=119.536ms
       ---
       at:
         line: 170
         column: 5
         file: test/async-await.js
       results:
         plan:
           start: 1
           end: 3
         count: 3
         pass: 2
         ok: false
         fail: 1
         time: 119.536
       source: |
         tap.test('sync-error', function (t) {
       ...
     # Subtest: async-error
         ok 1 - should be equivalent
         ok 2 - should be equivalent
         ok 3 - should be equivalent
         1..3
     ok 7 - async-error # time=125.421ms
     1..7
     # failed 1 of 7 tests
     # time=998.3ms
 not ok 4 - test/async-await.js # time=1120.491ms
   ---
   timeout: 30000
   file: test/async-await.js
   results:
     ok: false
     count: 7
     pass: 6
     fail: 1
     plan:
       start: 1
       end: 7
   exitCode: 1
   command: /home/iojs/build/workspace/citgm-smoker/nodes/ubuntu1804-64/smoker/bin/node
   arguments:
     - test/async-await.js
   ...
```
</details>

resolve@1.17.0 from https://ci.nodejs.org/view/Node.js-citgm/job/citgm-smoker/2523/nodes=ubuntu1804-64/testReport/(root)/citgm/resolve_v1_17_0/

<details>

```
not ok 190 assert/strict is a core module
...
not ok 210 dns/promises is a core module
...
not ok 250 stream/promises is a core module
...
not ok 258 timers/promises is a core module
```
</details>

**Comment by ljharb** [comment]
For tape, looks like node's error stacks now include a `at processImmediate (node:internal/timers:$LINE:$COL)` line where they didn't before - is that intentional and/or necessary?

**Comment by ljharb** [comment]
For resolve, v1.18.0 is now published; I believe that will pass the tests until new core modules are merged.

**Comment by BethGriggs** [comment]
> For tape, looks like node's error stacks now include a at processImmediate (node:internal/timers:$LINE:$COL) line where they didn't before - is that intentional and/or necessary?

I have just found the PR that introduced the change, see https://github.com/nodejs/node/pull/35498#issuecomment-704428305

**Comment by ljharb** [comment]
It's not the `node:` part that i'm noticing tho; it's the extra line entirely.

**Comment by richardlau** [comment]
> 
> 
> It's not the `node:` part that i'm noticing tho; it's the extra line entirely.

@ljharb It's the following line in tape's test that is affected by the `node:` part and no longer filtering out the line:
https://github.com/substack/tape/blob/898302b3e914c93b407088d36a224355b898bb0c/test/async-await.js#L184

```
> /\(internal\/timers.js:/.test('at processImmediate (internal/timers.js:462:21)')
true
> /\(internal\/timers.js:/.test('at processImmediate (node:internal/timers:462:21)')
false
>
```

**Comment by SimenB** [comment]
I'm assuming it's the same thing that made Jest fail: https://github.com/tapjs/stack-utils/issues/54

**Comment by richardlau** [comment]
> I'm assuming it's the same thing that made Jest fail: [tapjs/stack-utils#54](https://github.com/tapjs/stack-utils/issues/54) (that probably wouldn't affect tape, but possibly tap?)

Yes, looks like it. I've raised https://github.com/tapjs/stack-utils/pull/55.

**Comment by ljharb** [comment]
@richardlau thanks for clarifying, it’s now fixed in the repo but not yet published.

**Comment by SimenB** [comment]
Thanks @richardlau!

**Comment by lpinca** [comment]
The problem with `ws` is that `npm` no longer installs the [`utf-8-validate`](https://github.com/websockets/ws/blob/7.3.1/package.json#L55) dev dependency. Probably because it is also listed as [optional peer dependency](https://github.com/websockets/ws/blob/7.3.1/package.json#L41-L43).

I'm not sure how to fix this.

**Comment by lpinca** [comment]
`ws` issue is fixed in `npm@7.0.4`.

**Comment by targos** [comment]
Closing in favor of https://github.com/nodejs/citgm/issues/852
